app/src/main/java/org/splitbrain/giraffe/EventLoader.java

```diff
@@ -167,53 +272,30 @@ private EventRecord getEventRecord(SimpleIcalEvent event) {
         return record;
     }
 
+    private EventRecord getEventRecordJson(SimpleJsonEvent event) {
+        // mandatory fields
+        String uid = event.getId();
+        Date dateStart = event.getStarts();
 
-    /**
-     * always verify the host - don't check for certificate
-     *
-     * @link http://stackoverflow.com/a/9133562/172068
-     */
-    final static HostnameVerifier DO_NOT_VERIFY = new HostnameVerifier() {
-        @SuppressLint("BadHostnameVerifier")
-        public boolean verify(String hostname, SSLSession session) {
-            return true;
-        }
-    };
-
-    /**
-     * Trust every server - don't check for any certificate
-     *
-     * @link http://stackoverflow.com/a/9133562/172068
-     */
-    private static void trustAllHosts() {
-        // Create a trust manager that does not validate certificate chains
-        TrustManager[] trustAllCerts = new TrustManager[]{new X509TrustManager() {
-            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
-                return new java.security.cert.X509Certificate[]{};
-            }
-
-            @SuppressLint("TrustAllX509TrustManager")
-            public void checkClientTrusted(
-                    java.security.cert.X509Certificate[] chain, String authType)
-                    throws java.security.cert.CertificateException {
-            }
+        // optional fields
+        Date dateEnd = event.getEnds();
+        String location = event.getLocation();
+        String url = event.getUrl();
+        String description = event.getDescription();
+        String organizer = event.getSpeaker();
 
-            @SuppressLint("TrustAllX509TrustManager")
-            public void checkServerTrusted(
-                    java.security.cert.X509Certificate[] chain, String authType)
-                    throws java.security.cert.CertificateException {
-            }
-        }};
+        // create event
+        EventRecord record = new EventRecord();
+        record.id = uid;
+        record.title = event.getTitle();
+        record.starts = dateStart.getTime() / 1000;
+        if (dateEnd != null) record.ends = dateEnd.getTime() / 1000;
+        if (location != null) record.location = location;
+        if (organizer != null) record.speaker = organizer;
+        if (url != null) record.url = url;
+        if (description != null) record.description = description;
```

Comment:
This final cleanup is duplicated code again. It should probably be moved to it's own function.